java/runtime/src/main/java/io/ray/runtime/util/SystemConfig.java

```diff
@@ -0,0 +1,27 @@
+package io.ray.runtime.util;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import java.util.HashMap;
+
+/** The utility class to read system config from native code. */
+public class SystemConfig {
+
+  private static Gson gson = new Gson();
+
+  /// A cache to avoid duplicated reading via JNI.
+  private static HashMap<String, Object> cachedConfigs = new HashMap<>(1024);
+
+  public static synchronized Object get(String key) {
```

Comment:
Maybe it's better to return typed value instead of `Object`, like what we do in cpp code: 
```
num_retries = RayConfig::instance().raylet_client_num_connect_attempts();
```

Reply:
Because items are in different types.

Reply:
items in RayConfig also have different types :-)

Reply:
They are all code generated by macro, which is not able to be implemented in Java(Maybe lombok can work for it). We should define methods for every item.

Reply:
Okay that makes sense.